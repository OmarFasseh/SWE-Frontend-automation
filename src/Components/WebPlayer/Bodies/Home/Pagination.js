import React from 'react';
 /**
 * Pagination Method 
 * @extends Component
 * This is responsibe of the pagination Requirement that is used in Recent Activity
 * 
 * Pages is the total number of pages
 * @type {Number}
 * @param {Page} props 
 * 
 * Current Page that the user is on 
 * @type {Number}
 * @param {currentPage} props 
 * 
 * Next page 
 * @type {Number}
 * @param {NextPage} props 
 * 
 */

const Pagination = (props) => {
    /**
     * Number of links of pages that appear down on the pagination bar 
     * @type {Array}
     * @param {Pagelinks}  
     */
    let pageLinks = [] 
    
    for(let i = 1; i <= props.pages + 1; i++ ) {
        let active = props.currentPage == i ? 'active' : '';
        pageLinks.push(<li className={`waves-effect ${active}`} key={i} onClick={() => props.nextPage(i)}><a href="#!">{i}</a></li>)              
    }
    return ( 
            <div className="container">
                <div className="row">
                    <ul className="pagination">
                        { props.currentPage > 1 ? <li className="waves-effect" onClick={() => props.nextPage(props.currentPage - 1)}><a href="#!">Prev</a></li> : ''}
                        {  pageLinks }
                        { props.currentPage < props.pages + 1 ? <li className="waves-effect" onClick={() => props.nextPage(props.currentPage + 1)}><a href="#!">Next</a></li> : ''} 
                    </ul>
                </div>        
            </div>
    )
}

export default Pagination