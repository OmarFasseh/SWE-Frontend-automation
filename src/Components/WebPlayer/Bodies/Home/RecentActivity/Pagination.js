import React from 'react';
 /**
 * Pagination Method 
 * @extends Component
 * This is responsibe of the pagination Requirement that is used in Recent Activity
 * 
 * Pages is the total number of pages
 * @type {Number}
 * @param {Page} props 
 * 
 * Current Page that the user is on 
 * @type {Number}
 * @param {currentPage} props 
 * 
 * Next page 
 * @type {Number}
 * @param {NextPage} props 
 * 
 */

const Pagination = (props) => {
    /**
     * Number of links of pages that appear down on the pagination bar 
     * @type {Array}
     * @param {Pagelinks}  
     */
    let pageLinks = [] 
    
    for(let i = 1; i <= props.pages + 1; i++ ) {
        let active = props.currentPage == i ? 'active' : '';
        pageLinks.push(<li className={`waves-effect ${active}`} key={i} onClick={() => props.nextPage(i)}> <span class="notify-time">{i}</span></li>)              
    }
 
    return ( 
            <div className="container">
                <div className="row">
                    <ul className="pagination">
                        { props.currentPage > 1 ? <li className="waves-effect" onClick={() => props.nextPage(props.currentPage - 1)}><span class="notify-time">Prev</span></li> : ''}
                        {/* {console.log(props.currentPage)} */}
                        { props.currentPage < props.pages + 1 ? <li className="waves-effect" onClick={() => props.nextPage(props.currentPage + 1)}><span class="notify-time">Next</span></li> : ''} 
                        {/* {console.log(props.pages)} */}
                    </ul>
                </div>        
            </div>
    )
}

export default Pagination