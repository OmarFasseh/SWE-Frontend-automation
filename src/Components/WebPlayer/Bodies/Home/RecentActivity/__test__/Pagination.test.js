import React from "react";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import { create } from "react-test-renderer";
import { configure, shallow } from "enzyme";
import "jest-enzyme";
import "@testing-library/jest-dom/extend-expect";
import Pagination from "../Pagination";

Enzyme.configure({ adapter: new Adapter() });
configure({ adapter: new Adapter() });

describe("Recent Activity Test", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Pagination />);
  });


  it("Snapshot testing", () => {
    const PaginationTest = shallow(<Pagination />);
    expect(PaginationTest).toMatchInlineSnapshot(`
      <div
        className="container"
      >
        <div
          className="row"
        >
          <ul
            className="pagination"
          />
        </div>
      </div>
    `);
  });

  describe("Pagination component", () => {
    test("Matches the snapshot", () => {
      const PaginationCheck = create(<Pagination />);
      expect(PaginationCheck.toJSON()).toMatchSnapshot();
    });
  });
});
