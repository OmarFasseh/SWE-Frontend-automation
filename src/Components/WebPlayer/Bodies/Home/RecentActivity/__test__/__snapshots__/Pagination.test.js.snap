// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recent Activity Test Pagination component Matches the snapshot 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <ul
      className="pagination"
    >
      
      
    </ul>
  </div>
</div>
`;
